### PR TITLE
add path to route

### LINKED ISSUE #421: Follower requests never complete - ModSecurity off
I've successfully set up `@blog@shkspr.mobi` - it appears on the Fediverse. But people can't follow it.

ModSecurity is set to "off" in my cPanel. As per #341 and https://blog.rac.me.uk/2022/11/10/activitypub-for-wordpress-how-to-fix-modsecurity-to-make-it-work/#adding-the-new-mod-security-rule that should fix it. But it doesn't.

Screenshot showing ModSecurity off.

Are there any tests I can run to see what the problem is?

Site Health shows:
>  Author URL is not accessible ActivityPub
> Your author URL https://shkspr.mobi/blog/author/admin/ does not return valid JSON for application/activity+json. Please check if your hosting supports alternate Accept headers.

However, I am not using `admin` as the author; I've disabled the author in the settings and have a generic blog profile ID.

### PR DESCRIPTION
**Pull request by pfefferle**
fix #421

## Proposed changes:

* The current code ignores the 'path' when generating the signature string. This PR should fix the issue.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Install WordPress in a sub directory
* Install ActivityPub
* Try to follow your blog from Mastodon or a similar platform

### CODE REVIEW

**Comment by pfefferle** [comment]
@edent can you verify that this PR fixes your issue?

**Comment by edent** [comment]
🙌🙌🙌🙌🙌

I can now follow `@blog@shkspr.mobi`!

Will check in an hour or so if it pushes out the scheduled post to subscriber.

**Comment by pfefferle** [comment]
@mattwiebe might this break something on WP.com?

File: includes/class-signature.php [hunk 1]
```diff
@@ -238,6 +238,15 @@ public static function verify_http_signature( $request ) {
 			} else {
 				$route = '/' . rest_get_url_prefix() . '/' . ltrim( $request->get_route(), '/' );
 			}
+
+			// fix route for subdirectory installs
+			$path = wp_parse_url( get_home_url(), PHP_URL_PATH );
+			$path = trim( $path, '/' );
+
+			if ( $path ) {
```
Comment (mattwiebe) [inline includes/class-signature.php:246]: I'd maybe check instead for `! empty( $path )` here instead, although this would probably be ok.

**Review by mattwiebe — APPROVED** [review]
> wp_parse_url( get_home_url(), PHP_URL_PATH ); should be empty on public-api.wordpress.com/!?!

This shouldn't cause any trouble on dotcom or any non-subdirectory-based install. Looks good.

**Pull request merged**